R/ids_get.R

```diff
@@ -1,79 +1,119 @@
+#' easier to conduct cross-country debt analysis and monitoring.
 #'
-#' @param geographies A character vector representing the geographic codes
-#'  (e.g., "ZMB" for Zambia). This argument is required and cannot contain NA
-#'  values.
-#' @param series A character vector representing the series codes (e.g.,
-#'  "DT.DOD.DPPG.CD"). This argument is required and cannot contain NA values.
-#' @param counterparts A character vector representing counterpart areas (e.g.,
-#'  "all", "001"). This argument is required and cannot contain NA values
-#'  (default: "all").
-#' @param start_date An optional numeric value representing the starting year
-#'  (e.g., 2015). It must be greater than or equal to 1970. If not provided, the
-#'  entire time range is used.
-#' @param end_date An optional numeric value representing the ending year (e.g.,
-#'  2020). It must be greater than or equal to 1970 and cannot be earlier than
-#'  `start_date`. If not provided, the entire available time range is used.
-#' @param progress A logical value indicating whether to display a progress
-#'  message during the request process (default: `FALSE`). Must be either `TRUE`
-#'  or `FALSE`.
+#' @param geographies A character vector of geography identifiers representing
+#'   debtor countries and aggregates. Must use `geography_id` from
+#'   `ids_list_geographies()`:
+#'   * For individual countries, use ISO3C codes (e.g., "GHA" for Ghana)
+#'   * For aggregates, use World Bank codes (e.g., "LIC" for low income
+#'     countries)
+#'   The IDS database covers low and middle-income countries and related
+#'   aggregates only. Cannot contain NA values.
+#'
+#' @param series A character vector of debt statistics series identifiers that
+#'   must match the `series_id` column from `ids_list_series()`. Each series
+#'   represents a specific debt statistic (e.g., "DT.DOD.DECT.CD" for total
+#'   external debt stocks, "DT.TDS.DECT.CD" for debt service payments). Cannot
+#'   contain NA values.
+#'
+#' @param counterparts A character vector of creditor identifiers that must
+#'   match the `counterpart_id` column from `ids_list_counterparts()`. The
+#'   default "WLD" returns aggregated global totals across all creditors.
+#'   Common options:
+#'   * "WLD" - World total (aggregated across all creditors)
+#'   * "all" - Retrieve data broken down by all creditors
+#'   * Individual creditors use numeric codes (e.g., "730" for China)
+#'   * Special creditors have text codes (e.g., "907" for IMF, "BND" for
+#'     bondholders)
+#'   Cannot contain NA values.
+#'
+#' @param start_date A numeric value representing the starting year (default:
+#'   2000). Must be >= 1970. The default focuses on modern data while reducing
+#'   data volume. For historical analysis, explicitly set to 1970.
+#'
+#' @param end_date A numeric value representing the ending year (default: NULL).
+#'   Must be >= 1970 and cannot be earlier than start_date. If NULL, returns
```

Comment:
We actually could remove this constraint, replace any sub-1970 year with 1970, and just raise a warning flag to let the user know we did so.